src/test/java/command/DeadlineCommandTest.java

```diff
@@ -0,0 +1,34 @@
+package command;
+
+import duke.Ui;
+import duke.commands.DeadlineCommand;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.tasks.Deadline;
+import duke.tasks.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DeadlineCommandTest {
+    TaskList tasks;
+    Ui ui;
+    Storage storage;
+
+    @BeforeEach
+    public void init() {
+        this.tasks = new TaskList();
+        this.ui = new Ui();
+        this.storage = new Storage("./data/");
+    }
+
+    @Test
+    public void shouldAddDeadline() {
+        DeadlineCommand demoDeadlineCommand = new DeadlineCommand("Demo /by 2002-02-12 19:00");
+        demoDeadlineCommand.execute(this.tasks, this.ui, this.storage);
+        Task actual = this.tasks.showTask(0);
+        Deadline expected = new Deadline("Demo","2002-02-12 19:00");
+        assertEquals(expected, actual);
+    }
```

Comment:
Is it a good idea to test for errors thrown? So that you can check if the right error is thrown for the right edge cases.